### PR TITLE
chore(ui): update UI vue version

### PR DESCRIPTION
**Pull request by dammy001**

### CODE REVIEW

File: packages/ui/client/components/Modal.vue [hunk 1]
```diff
@@ -1,15 +1,11 @@
 <script setup lang='ts'>
 const props = withDefaults(defineProps<{
```
[the remark this answers is not in the document]
Comment (userquin) [reply packages/ui/client/components/Modal.vue:2]: maybe we need to review `withDefaults` usages

[the remark this answers is not in the document]
Comment (dammy001) [reply packages/ui/client/components/Modal.vue:2]: Yeah...

**Comment by userquin** [comment]
@dammy001 can you try to run `nr build` from root? vitest package being built after ui package...

**Comment by dammy001** [comment]
> @dammy001 can you try to run `nr build` from root? vitest package being built after ui package...
> 
> EDIT: I've it fixed on my local, updating PR..

@userquin okay i'll update

**Comment by dammy001** [comment]
@userquin Okay, Thanks.

**Comment by userquin** [comment]
The PR LGTM, just resolve conflicts once green: on my local build works, tested new UI with `test/core`, and  running `nr test`and `nr test:ci` both ok

**Review by userquin — APPROVED** [review]

**Comment by dammy001** [comment]
> The PR LGTM, just resolve conflicts once green: on my local build works, tested new UI with `test/core`, and running `nr test`and `nr test:ci` both ok

@userquin tested the same also. LGTM

**Comment by dammy001** [comment]
@sheremet-va checked this?

File: packages/vitest/src/runtime/rpc.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import {
   getSafeTimers,
 } from '@vitest/utils'
+import type { BirpcReturn } from 'birpc'
 import { getWorkerState } from '../utils/global'
+import type { RuntimeRPC } from '#types'
```
Comment (sheremet-va) [inline packages/vitest/src/runtime/rpc.ts:6]: You can' access `#types` not in packages/ui

**Review by sheremet-va — CHANGES_REQUESTED** [review]
Why are there changes in places that are not related to UI?

File: packages/vitest/src/runtime/rpc.ts (hunk 2 above)
Comment (dammy001) [reply packages/vitest/src/runtime/rpc.ts:6]: @sheremet-va these changes are related to fixes for CI made by @userquin

**Comment by userquin** [comment]
> Why are there changes in places that are not related to UI?

do you mean why we updated vitest default and rpc modules? TS complains, I cannot build repo without those changes

**Comment by sheremet-va** [comment]
> > Why are there changes in places that are not related to UI?
> 
> do you mean why we updated vitest default and rpc modules? TS complains, I cannot build repo without those changes

Well, there are no problems in the main branch. I can't approve PR that has unrelated changes

File: packages/vitest/src/defaults.ts [hunk 3]
```diff
@@ -96,4 +96,4 @@ const config = {
   slowTestThreshold: 300,
 }
 
-export const configDefaults: Required<Pick<UserConfig, keyof typeof config>> = Object.freeze(config)
+export const configDefaults = Object.freeze(config as Required<UserConfig>)
```
Comment (sheremet-va) [inline packages/vitest/src/defaults.ts:99]: This actually breaks `configDefaults` types. It should ONLY reference keys that exist in default config

Comment (userquin) [reply packages/vitest/src/defaults.ts:99]: I cannot review it rn, I will try to fix it tmr...

File: packages/vitest/src/runtime/rpc.ts (hunk 2 above)
Comment (sheremet-va) [reply packages/vitest/src/runtime/rpc.ts:6]: It doesn't mean that it's wrong

Comment (dammy001) [reply packages/vitest/src/runtime/rpc.ts:6]: @userquin i get you.

**Comment by userquin** [comment]
@dammy001 can you resolve conflicts?

**Comment by dammy001** [comment]
Okay. @userquin

**Comment by dammy001** [comment]
@userquin conflict resolved. @sheremet-va please check.

**Comment by userquin** [comment]
@dammy001 can you move `defineOptions` to be the first after static imports in `TaskTree` and `TasksList` components?

**Comment by userquin** [comment]
ok, the linter will move `defineOptions`  to the script setup bottom

**Comment by userquin** [comment]
@dammy001 have you merged main? the lock file version is 6.1

**Comment by dammy001** [comment]
@userquin yeah. Merged with main already. Your local branch updated?

**Comment by userquin** [comment]
ok, it seems you don't have installed pnpm 8.6.6, fixing lock file and linting...

**Comment by dammy001** [comment]
@sheremet-va LGTM

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**